tests/e2e/vsphere_volume_expansion.go

```diff
@@ -412,6 +421,470 @@ var _ = ginkgo.Describe("Volume Expansion Test", func() {
+
+		var wg sync.WaitGroup
+		wg.Add(3)
+		go resize(client, pvclaim, currentPvcSize, newSize1, &wg)
+		go resize(client, pvclaim, currentPvcSize, newSize3, &wg)
+		go resize(client, pvclaim, currentPvcSize, newSize2, &wg)
+		wg.Wait()
+
+		pvclaim, err = client.CoreV1().PersistentVolumeClaims(namespace).Get(ctx, pvclaim.Name, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		gomega.Expect(pvclaim).NotTo(gomega.BeNil())
+
+		ginkgo.By("Waiting for file system resize to finish")
+		pvclaim, err = waitForFSResize(pvclaim, client)
+		framework.ExpectNoError(err, "while waiting for fs resize to finish")
+
+		pvcConditions := pvclaim.Status.Conditions
+		expectEqual(len(pvcConditions), 0, "pvc should not have conditions")
+
+		ginkgo.By("Verify filesystem size for mount point /mnt/volume1")
+		fsSize, err := getFSSizeMb(f, pod)
+		framework.Logf("FileSystemSize after PVC resize %d mb , FileSystemSize Before PVC resize %d mb ", fsSize, originalFSSize)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		// Filesystem size may be smaller than the size of the block volume
+		// so here we are checking if the new filesystem size is greater than
+		// the original volume size as the filesystem is formatted for the
+		// first time
+		if fsSize < originalFSSize {
+			framework.Failf("error updating filesystem size for %q. Resulting filesystem size is %d", pvclaim.Name, fsSize)
+		}
+		ginkgo.By("File system resize finished successfully")
+
+		pvcsize := pvclaim.Spec.Resources.Requests[v1.ResourceStorage]
+		actualSizeOfPVC := sizeInMb(pvcsize.DeepCopy())
+		expectedSize := int64(10240)
+		framework.Logf("Actual size after converting to MB : %d", actualSizeOfPVC)
+		if actualSizeOfPVC != expectedSize {
+			framework.Failf("Received wrong disk size after volume expansion. Expected: %d Actual: %d", expectedSize, actualSizeOfPVC)
+
+		}
+
+	})
+
+	/*
+		Verify online volume expansion on Shared VVOL datastore
+		1. Create StorageClass with allowVolumeExpansion set to true on a shared VVOL datastore.
+		2. Create PVC which uses the StorageClass created in step 1.
+		3. Wait for PV to be provisioned.
+		4. Wait for PVC's status to become Bound and note down the size
+		5. Create a POD using the above created PVC
+		6. Modify PVC's size to trigger online volume expansion
+		7. verify the PVC status will change to "FilesystemResizePending". Wait till the status is removed
+		8. Verify the resized PVC by doing CNS query
+		9. Make sure data is intact on the PV mounted on the pod
+		10.  Make sure file system has increased
+	*/
+	ginkgo.It("[csi-block-vanilla] Verify online volume expansion on shared VVOL datastore", func() {
+		ginkgo.By("Invoking Test for Volume Expansion")
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+
+		sharedVVOLdatastoreURL := os.Getenv(envSharedVVOLDatastoreURL)
+		if sharedVVOLdatastoreURL == "" {
+			ginkgo.Skip("env SHARED_VVOL_DATASTORE_URL is missing")
```

Comment:
We shouldn't skip tests. We should add custom strings in `ginkgo.It` so that users can decide to skip such tests in command line arguments. And also this test should not run if users  provide `csi-block-vanilla` as the ginkgo focus

Reply:
As of now in ginkgo.It we are classifying the tests based on the testbed flavour.  This test is specific to datastore (this belongs to all flavour). If the test bed has this datastore set it will run the test , If it does not have the datastore it will skip the test. 

To give more context on why we are skipping the test , I will add the detail description instead of just saying "env SHARED_VVOL_DATASTORE_URL is missing"